File: add_advanced_data.py
import json
import urllib.request
import re

JS_FILE = '/Users/pochoco/Desktop/원소주기율표/elements_data.js'
URL = 'https://raw.githubusercontent.com/Bowserinator/Periodic-Table-JSON/master/PeriodicTableJSON.json'

print("Downloading advanced data from Bowserinator's Periodic-Table-JSON...")
try:
    with urllib.request.urlopen(URL) as url:
        data = json.loads(url.read().decode())
except Exception as e:
    print(f"Failed to download data: {e}")
    exit(1)

elements_dict = {}
for el in data['elements']:
    elements_dict[el['number']] = {
        'density': el.get('density', None),
        'electronegativity': el.get('electronegativity_pauling', None),
        'electron_config': el.get('electron_configuration_semantic', ''),
        'shells': ', '.join(map(str, el.get('shells', [])))
    }

print("Reading elements_data.js...")
with open(JS_FILE, 'r', encoding='utf-8') as f:
    js_content = f.read()

# Extract JSON string from JS
match = re.search(r'const elementsData = (\[.*?\]);', js_content, re.DOTALL)
if not match:
    print("Could not find elementsData array in JS file.")
    exit(1)

json_str = match.group(1)
# Clean up JS specific syntax if needed (like trailing commas, missing quotes around keys).
# Our elements_data.js is already valid JSON, but keys might not be quoted if it's raw JS.
# Assuming it's well-formed enough or we can use demjson, but let's try standard json if possible.
# Actually, the previous add_temp_data.py handled it with regex or by rewriting.
# Let's use a simpler approach: evaluating it.

File: add_advanced_data.js
const fs = require('fs');
const https = require('https');

const JS_FILE = '/Users/pochoco/Desktop/원소주기율표/elements_data.js';
const URL = 'https://raw.githubusercontent.com/Bowserinator/Periodic-Table-JSON/master/PeriodicTableJSON.json';

console.log("Downloading advanced data...");

https.get(URL, (res) => {
    let body = '';
    res.on('data', chunk => body += chunk);
    res.on('end', () => {
        const data = JSON.parse(body);
        const elementsDict = {};
        
        data.elements.forEach(el => {
            elementsDict[el.number] = {
                density: el.density || null,
                electronegativity: el.electronegativity_pauling || null,
                electron_config: el.electron_configuration_semantic || '',
                shells: el.shells ? el.shells.join(', ') : '',
                discovered_by: el.discovered_by || '',
                named_by: el.named_by || '',
                appearance: el.appearance || '',
                molar_heat: el.molar_heat || null,
                electron_affinity: el.electron_affinity || null,
                ionization_energies: el.ionization_energies ? el.ionization_energies[0] : null,
                summary: el.summary || '',
                spectral_img: el.spectral_img || ''
            };
        });

        console.log("Reading elements_data.js...");
        let jsContent = fs.readFileSync(JS_FILE, 'utf-8');
        
        // Extract array
        const match = jsContent.match(/const elementsData = (\[[\s\S]*\]);/);
        if (!match) {
            console.error("Could not find elementsData array in JS file.");
            process.exit(1);
        }
        
        // Eval to parse as JS object
        let elementsData;
        eval('elementsData = ' + match[1]);
        
        // Update data
        elementsData.forEach(el => {
            const advanced = elementsDict[el.num];
            if (advanced) {
                el.density = advanced.density;
                el.electronegativity = advanced.electronegativity;
                el.electron_config = advanced.electron_config;
                el.shells = advanced.shells;
                el.discovered_by = advanced.discovered_by;
                el.named_by = advanced.named_by;
                el.appearance = advanced.appearance;
                el.molar_heat = advanced.molar_heat;
                el.electron_affinity = advanced.electron_affinity;
                el.ionization_energies = advanced.ionization_energies;
                el.summary = advanced.summary;
                el.spectral_img = advanced.spectral_img;
            }
        });
        
        // Serialize back
        const newJsContent = jsContent.replace(
            match[1], 
            JSON.stringify(elementsData, null, 2)
        );
        
        fs.writeFileSync(JS_FILE, newJsContent, 'utf-8');
        console.log("Successfully injected advanced data into elements_data.js!");
    });
}).on('error', (e) => {
    console.error(e);
});
